code/script.js

```diff
@@ -253,91 +257,136 @@ const setSecret = () => {
-  // We also need a variable that stores the actual value of the question we've selected.
-
+  
+const value = questions.value
   if (category === 'hair color') {
     currentQuestion = {
       attribute: 'hairColor',
-      // value: ,
-      // 👆 add the value from the input here
-      category: category,
+      value,
+      category,
     }
   } else if (category === 'eye color') {
-    // Set this up your self
+    currentQuestion = {
+      attribute: 'eyeColor',
+      value,
+      category,
+    }
   } else if (category === 'accessories') {
     currentQuestion = {
-      //attribute: ,
-      // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
-      value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
-      category: category,
+      attribute: value,
+      value: true,
+      category,
     }
   } else if (category === 'other') {
-    // Set this up your self (should be same structure as above)
+    currentQuestion = {
+      attribute: value,
+      value: true,
+      category,
+    }
   }
 }
 
 // This function should be invoked when you click on 'Find Out'.
 const checkQuestion = () => {
-  // Compare the currentQuestion with the secret person.
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
+  const keep = currentQuestion.value === secret[currentQuestion.attribute]
+
+  filterCharacters(keep)
 }
 
 // It'll filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
-  // Show the correct alert message for different categories
-  if (group === 'accessories') {
+  const {attribute, value, category} = currentQuestion
+  
+  if (category === 'accessories') {
     if (keep) {
       alert(
-        `Yes, the person wears ${attribute}! Keep all that wears ${attribute}`
+        `Good Job! The person wears ${attribute}! I'm keeping all that wear ${attribute} for you.`
       )
     } else {
       alert(
-        `No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}`
+        `Naah.. The person doesn't wear ${attribute}! I'm getting rid of all that wear ${attribute}.`
       )
     }
-  } else if (group === 'other') {
-    // Similar to the one above
-  } else {
-    if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
+  } else if (category === 'eye color'){
+    if (keep){
+      alert(
+        `Yes, the person has pretty ${value} eyes! I'm keeping all that have those pretty ${value} eyes.`
+      )
     } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
+      alert(
+        `Nope. The person doesn't have ${value} eyes! I'm removing all that have ${value} eyes. Nothing personal - just for you.`
```

Comment:
I love what you did with the wording, it's definitely engaging! I played it twice and I actually had fun with your alerts. 😄